solidity.js

```diff
@@ -364,49 +440,44 @@ function hljsDefineSolidity(hljs) {
                 contains: [
                     hljs.C_LINE_COMMENT_MODE,
                     hljs.C_BLOCK_COMMENT_MODE,
-                    { //the actual *block* in the assembly section
+                    hljs.inherit(SOL_ASSEMBLY_ENVIRONMENT, { //the actual *block* in the assembly section
                         begin: '{', end: '}',
                         endsParent: true,
-                        keywords: SOL_ASSEMBLY_KEYWORDS,
-                        lexemes: SOL_ASSEMBLY_LEXEMES_RE,
-                        contains: [
-                            SOL_APOS_STRING_MODE,
-                            SOL_QUOTE_STRING_MODE,
-                            HEX_APOS_STRING_MODE,
-                            HEX_QUOTE_STRING_MODE,
-                            hljs.C_LINE_COMMENT_MODE,
-                            hljs.C_BLOCK_COMMENT_MODE,
-                            SOL_NUMBER,
-                            SOL_ASSEMBLY_MEMBERS,
-                            SOL_ASSEMBLY_MEMBERS_OLD,
-                            { //block within assembly; note the lack of endsParent
+                        contains: SOL_ASSEMBLY_ENVIRONMENT.contains.concat([
+                            hljs.inherit(SOL_ASSEMBLY_ENVIRONMENT, { //block within assembly
                                 begin: '{', end: '}',
-                                keywords: SOL_ASSEMBLY_KEYWORDS,
-                                lexemes: SOL_ASSEMBLY_LEXEMES_RE,
-                                contains: [
-                                    SOL_APOS_STRING_MODE,
-                                    SOL_QUOTE_STRING_MODE,
-                                    HEX_APOS_STRING_MODE,
-                                    HEX_QUOTE_STRING_MODE,
-                                    hljs.C_LINE_COMMENT_MODE,
-                                    hljs.C_BLOCK_COMMENT_MODE,
-                                    SOL_NUMBER,
-                                    SOL_ASSEMBLY_MEMBERS,
-                                    SOL_ASSEMBLY_MEMBERS_OLD,
-                                    'self'
-                                ]
-                            }
-                        ]
-                    }
+                                contains: SOL_ASSEMBLY_ENVIRONMENT.contains.concat(['self'])
+                            })
+                        ])
+                    })
                 ]
             }
         ],
         illegal: /#/
     };
 }
 
+function hljsDefineYul(hljs) {
+
+    var YUL_KEYWORDS = {
+        keyword: SOL_ASSEMBLY_KEYWORDS.keyword + ' ' +
+            'object code data',
```

Comment:
Exactly; trying to do a proper grammar would I think be too much work.  Yul's grammar is simple enough that it might actually be possible, but since we don't have that with Solidity, I didn't bother here either.  So I'm going to leave this alone unless you really think it should change.